zerver/forms.py

```diff
@@ -100,8 +100,10 @@ def clean_full_name(self):
     def clean_realm_subdomain(self):
         # type: () -> str
         subdomain = self.cleaned_data['realm_subdomain']
-        if not subdomain:
-            return ''
+        if not subdomain and not self.realm_creation:
```

Comment:
@timabbott This can be `if not self.realm_creation:` but `./tools/test-backend zerver.tests.test_signup.UserSignUpTest.test_registration_through_ldap breaks in that case. This test passes subdomain and realm name data to user registration which I am not sure why. One explanation I was able to come up with was to make the form invalid(as "zulip" is passed as subdomain value which makes the form invalid as it is used) which is kind of necessary to for the test flow. 
```py
   result = self.submit_reg_form_for_user(email,
                                                       password,
                                                       full_name="Non LDAP Full Name",
                                                       realm_name=realm_name,
                                                       realm_subdomain=subdomain,
                                                       # Pass HTTP_HOST for the target subdomain
                                                       HTTP_HOST=subdomain + ".testserver")
```